scripts/run_local.sh

```diff
@@ -0,0 +1,170 @@
+if [ "$?" -ne "0" ]; then 
+  echo "Unable to write the service-ca.crt data for ${BROKER_SVC_ACCT_SECRET_NAME} to: ${SVC_ACCT_CA_CRT}"
+  exit 1 
+fi 
+echo "Service Account: ca.crt"
+echo -e "Wrote \n${SVC_ACCT_CA_CRT_DATA}\n to: ${SVC_ACCT_CA_CRT}\n"
+
+###
+# Fetch the token for the service account
+###
+if [ ! -d $SVC_ACCT_TOKEN_DIR ]; then
+  echo "Please create the directory: ${SVC_ACCT_TOKEN_DIR}"
+  echo "Ensure your user can write to it."
+  exit 1
+fi
+BROKER_SVC_ACCT_TOKEN=`oc get secret ${BROKER_SVC_ACCT_SECRET_NAME} -n ${ASB_PROJECT} -o json | jq -c '.data["token"]'`
+BROKER_SVC_ACCT_TOKEN=( $(eval echo ${BROKER_SVC_ACCT_TOKEN[@]}) )
+BROKER_SVC_ACCT_TOKEN=`echo ${BROKER_SVC_ACCT_TOKEN} | base64 --decode`
+###
+# Note:
+# It is important we do __not__ append the trailing newline in the token file
+# Go's ioutil module will read in the newline as part of the token which breaks it...and causes confusion tracking down
+###
+echo -n "${BROKER_SVC_ACCT_TOKEN}" &> $SVC_ACCT_TOKEN_FILE
+if [ "$?" -ne 0 ]; then 
+  echo "Unable to write token to $SVC_ACCT_TOKEN_FILE"
+  exit 1
+fi
+echo "Service Account: token"
+echo -e "Wrote \n${BROKER_SVC_ACCT_TOKEN}\n to: ${SVC_ACCT_TOKEN_FILE}\n"
+
+# Kill any running broker pods
+oc scale deployments asb --replicas 0 -n ${ASB_PROJECT}
+# Wait for asb pod to be destroyed
+oc get pods -n ${ASB_PROJECT} | grep asb 
+while [ "$?" -ne 1 ]; do 
+  echo "Waiting for asb deployment to scale down"
+  sleep 5
+  oc get pods -n ${ASB_PROJECT} | grep asb 
+done 
+
+oc delete endpoints asb -n ${ASB_PROJECT}
+oc delete service asb  -n ${ASB_PROJECT}
+oc delete route asb-etcd -n ${ASB_PROJECT}
+# Process required changes for local development
+oc process -f ${TEMPLATE_LOCAL_DEV} -n ${ASB_PROJECT} -p BROKER_IP_ADDR=${BROKER_IP_ADDR} | oc create -n ${ASB_PROJECT} -f - 
+
+echo "Sleeping for a few seconds to avoid issues with broker not being able to talk to etcd."
+echo "Appears like there is a delay of when we create the asb-etcd route and when it is available for use"
+sleep 5 
+
+etcd_route=`oc get route asb-etcd -n ${ASB_PROJECT} -o=jsonpath=\'\{.spec.host\}\'`
+echo "etcd route is at: ${etcd_route}"
+
+if [ -z "$DOCKERHUB_USERNAME" ]; then
+  echo "Please set the environment variable DOCKERHUB_USERNAME and re-run"
+  exit 1 
+fi 
+if [ -z "$DOCKERHUB_PASSWORD" ]; then
+  echo "Please set the environment variable DOCKERHUB_PASSWORD and re-run"
+  exit 1 
+fi 
+
+cat << EOF  > ${GENERATED_BROKER_CONFIG}
+---
+registry:
+  name: dockerhub
+  url: https://registry.hub.docker.com
+  user: ${DOCKERHUB_USERNAME}
+  pass: ${DOCKERHUB_PASSWORD}
+  org: ansibleplaybookbundle
+dao:
+  etcd_host: ${etcd_route}
+  etcd_port: 80
+log:
+  logfile: /tmp/ansible-service-broker-asb.log
+  stdout: true
+  level: debug
+  color: true
+openshift: {}
+broker:
+  dev_broker: true
+  launch_apb_on_bind: false
+  recovery: true
+  output_request: true
+EOF
+
+${BROKER_CMD} -c ${GENERATED_BROKER_CONFIG} 
```

Comment:
Separating the targets is not a bad idea; setup would basically reconfigure everything for a true local setup, and then run literally just runs a binary. changes, make run, Ctl+C, changes, make run loop is about as short as it can get, it's limited to binary build times.